sqlobject/src/main/java/org/jdbi/v3/sqlobject/SqlObject.java

```diff
@@ -82,13 +94,15 @@ public Object intercept(Object o, Method method, Object[] objects, MethodProxy m
         }
 
         final SqlObject so = new SqlObject(buildHandlersFor(sqlObjectType), handle);
-        return (T) f.newInstance(new MethodInterceptor()
-        {
-            @Override
-            public Object intercept(Object o, Method method, Object[] objects, MethodProxy methodProxy) throws Throwable
+        return (T) f.newInstance(new Callback[] {new MethodInterceptor()
             {
-                return so.invoke(o, method, objects, methodProxy);
-            }
+                @Override
+                public Object intercept(Object o, Method method, Object[] objects, MethodProxy methodProxy) throws Throwable
```

Comment:
Yeah, duplicating this sucks.  I chose to keep the cleanup of this to a separate refactoring PR rather than mixing it in with the feature request.